legacy/reqcounter/reqcounter_test.go

```diff
@@ -0,0 +1,191 @@
+	if rc.EnableCounting == false {
+		fmt.Println("Init did not enable counting")
+		t.Fail()
+	}
+	// Ensure counters are created.
+	if len(rc.Monitor.Counters) == 0 {
+		fmt.Println("Init did not create any request counters")
+		t.Fail()
+	}
+	// Ensure at least one counter uses the MeasurementWindow
+	found := false
+	for _, counter := range rc.Monitor.Counters {
+		if counter.Interval == rc.MeasurementWindow {
+			found = true
+			break
+		}
+	}
+	if !found {
+		fmt.Println("No request counters are using the 10min Interval: MeasurementWindow")
+		t.Fail()
+	}
+}
+
+func TestIncrement(t *testing.T) {
+	// Create several unique request counts.
+	requests := []uint64{0, 9, 2839}
+	// Create counters which use these request counts and
+	// populate the Monitor global variable.
+	rc.Monitor = &rc.NetworkMonitor{
+		Counters: []*rc.RequestCounter{
+			{
+				Mutex:    sync.Mutex{},
+				Requests: requests[0],
+				Since:    time.Now(),
+				Interval: time.Second,
+			},
+			{
+				Mutex:    sync.Mutex{},
+				Requests: requests[1],
+				Since:    time.Now(),
+				Interval: time.Second,
+			},
+			{
+				Mutex:    sync.Mutex{},
+				Requests: requests[2],
+				Since:    time.Now(),
+				Interval: time.Second,
+			}},
+	}
+	// Ensure that modifying counters can only occur when counting is enabled.
+	rc.EnableCounting = false
+	rc.Increment()
+	for i, counter := range rc.Monitor.Counters {
+		if counter.Requests != requests[i] {
+			fmt.Println("Counters were modified while counting was disabled")
+			t.Fail()
+		}
+	}
+	// Ensure the Increment function actually increments each counter's requests field.
+	rc.EnableCounting = true
+	rc.Increment()
+	for i, counter := range rc.Monitor.Counters {
+		if counter.Requests != requests[i]+1 {
+			fmt.Printf("Counters were not incremented correctly. Expected: %d Got: %d\n", requests[i]+1, counter.Requests)
+			t.Fail()
+		}
+	}
+}
+
+func TestFlush(t *testing.T) {
+	// Create a local invocation of time.
+	currentTime := time.Now()
+	counter := &rc.RequestCounter{
+		Mutex:    sync.Mutex{},
+		Requests: 100,
+		Since:    currentTime,
+		Interval: time.Second,
+	}
+	// Mock the logrus.Debug function.
+	debugCalls := 0
+	rc.Debug = func(args ...interface{}) {
+		debugCalls++
+	}
+	// Flush the request counter.
+	counter.Flush()
+	// Ensure logrus.Debug was called.
+	if debugCalls != 1 {
+		fmt.Println("Flush failed to trigger a debug statement.")
+		t.Fail()
+	}
+	// Ensure the counter is reset, where time advances and the requests are zeroed.
+	counter.Flush()
+	if counter.Requests != 0 || !currentTime.Before(counter.Since) {
+		fmt.Println("Calling Flush did not reset the counter.")
+		t.Fail()
+	}
+}
+
+func TestLog(t *testing.T) {
+	// Create two counters with different logging intervals. During testing, only one
+	// of the counters should fire logrus.Debug statements. To determine which counter
+	// was logged, we can use the Requests number to identify them.
+	unexpected := uint64(80808)
```

Comment:
Instead of unexpected and expected, use `shouldBeInvalid` and `shouldBeValid`.

I think there should be a NewRequestCounter method because it's hard to grok the values inside Counters.

Reply:
This test has since been completely refactored. Resolving.